hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -721,18 +747,49 @@
 
             Write a 32-byte key and a 16-byte plain text to the AES registers and trigger the AES
             computation to start. Wait for the AES operation to complete by polling the status
-            register (or interrupt if available). Check the digest registers for correctness against
-            the expected digest value.
+            register. Check the digest registers for correctness against the expected digest value.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_aes_entropy
+      desc: '''Verify the AES entropy input used by the internal PRNGs.
+
+            - Write the initial key share, IV and data in CSRs (known combinations).
+            - Write the PRNG_RESEED bit to reseed the internal state of the PRNG.
+            - Poll the status idle bit to ensure reseed operation is complete.
+            - Trigger the AES operation to run and wait for it to complete.
+            - Check the digest against the expected value.
+            - Write the KEY_IV_DATA_IN_CLEAR and DATA_OUT_CLEAR trigger bits to 1 and wait for it to
+              complete by polling the status idle bit.
+            - Read back the data out CSRs - they should all read garbage values.
+            - Assertion check verifies that the internal states (data_in, key share and IV are also
+              garbage, i.e. different from the originally written values.
+            - Assertion checks proves that all interfaces are connected across AST RNG, ES, CSRNG,
+              EDN and AES.
+            - TODO: Investigate if we can inject predictable RNG at AST so that the PRNG generated
+              data is predictable and verifiable.
```

Comment:
Since this function exists in entropy_src (lfsr mode vs. rng mode), maybe it would be to do this test with this mode on. Also, the entropy will be generated much faster than in normal rng mode.  Then another dedicated test can be added to check just the rng to es connections.